File: algorithms/sort/selection-sort/selection-sort.go
package main

func SelectionSort(array []int) []int {
	currentIndex := 0;
	for currentIndex < len(array)-1{
		minIndex := currentIndex
		for i := currentIndex+1; i < len(array)-1; i++ {
			if(array[minIndex] > array[i]) {
				minIndex = i
			}
		}
		swap(currentIndex,minIndex,array)
		currentIndex++
	}
	return array
}


func swap(i,j int, array []int)  {
		array[i] ,array[j] = array[j] , array[i]
}

File: algorithms/sort/selection-sort/selection-sort.java
import java.util.*;

class Program {
  public static int[] selectionSort(int[] array) {
    int currentIndex = 0;
    while (currentIndex < array.length-1) {
        int minIndex = currentIndex;
        for(int i = currentIndex+1; i < array.length;i++){
          if(array[minIndex] > array[i]) minIndex = i;
        }
        swap(currentIndex, minIndex, array);
        currentIndex++;
    }
    return array;
  }
  public static void swap(int i, int j, int[] array ){
    int temp = array[i];
    array[i] = array[j];
    array[j] = temp;
  }
}
